beancount_dkb/ec.py

```diff
@@ -222,11 +223,14 @@ def _update_meta(self, meta: Dict[str, str]):
                 # So when setting the balance date, we add a timedelta
                 # of 1 day to the original value to make the balance
                 # assertions work.
-
+                value_match = re.match("([0-9.]+(,[0-9]+)?).(EUR|€)", value.value)
+                if not value_match:
+                    raise ValueError(f"Can't parse {value.value}")
+                value_string = value_match.group(1)
```

Comment:
Personally, I would also be fine if we skip this validation entirely and just do something like the following for the sake of simplicity:

```python
if value.value.endswith(" EUR"):
  ...
elif value.value.endswith(" €"):
  ...
```

Reply:
The CSV I exported a few days ago will fail with this, because of the non-breaking-space before the `€`.

In the end I think there are much better ways to convert a float string in some locale to an actual float. But it's your call.

Reply:
I think those are two different topics: one is removing the currency suffix, and two is getting a float out.

Getting a float out is done inside `fmt_number_de`. For removing the currency suffix, we can also strip out the space(s) before passing it on to `fmt_number_de`.

```python
if value.value.endswith("€"):
  suffix = "€"
elif value.value.endswith("EUR")
  suffix = "EUR"
else:
  raise ValueError(...)

value_str = value.value.rstrip(suffix).strip()
```

If that continues to work with the latest CSV export, I would prefer that slightly, as it'll make future maintenance work for me easier, as the float conversion is then validated/done in one single function.

Reply:
Well how about dropping all the string mangling and use python to do both?

```
value_string = "1.912,39 EUR" # or value_string = "1.912,39 €"
import locale
locale.setlocale(locale.LC_ALL, "de_DE.utf8")
value = locale.atof(value_string.split()[0])
```

This is just an example and would have to be incorporated into your code.

Reply:
I think that was how it used to be in the beginning. But afterwards I had a look at Python's documentation on the `locale` module and it actually has [quite a few warnings](https://docs.python.org/3/library/locale.html#background-details-hints-tips-and-caveats) against using it in libraries. Based on that, I decided to remove it in favor of what it is now.

So if we set the locale to something (and even if restore it afterwards) it'll potentially interfere with eg. Beancount itself. As we only have one locale to handle anyway, string mangling seems like the simplest option to go for.

Reply:
Fair enough. So what about just leave `value.value.split()[0]` to strip off the suffix of `€` or `EUR` ?

Reply:
Also works for me!